sklearn/cluster/tests/test_k_means.py

```diff
@@ -136,7 +136,7 @@ def test_relocate_empty_clusters(array_constr):
 @pytest.mark.parametrize("distribution", ["normal", "blobs"])
 @pytest.mark.parametrize("array_constr", [np.array, sp.csr_matrix],
                          ids=["dense", "sparse"])
-@pytest.mark.parametrize("tol", [1e-2, 1e-4, 1e-8])
+@pytest.mark.parametrize("tol", [1e-2, 1e-4, 1e-8, 0])
```

Comment:
Maybe also add a small almost-zero value of tol that causes the same problems as using 0.